apt/preferences.sls

```diff
@@ -14,6 +14,7 @@
     - mode: '0644'
     - user: root
     - group: root
+    - replace: False
```

Comment:
According to the docs, using `replace: False` will make the formula respect the file contents no matter what they are, which can potentially lead to unexpected results when mixed with the preferences being set via the formula.

> replace
> : True
>     If set to False and the file already exists, the file will not be modified even if changes would otherwise be made. Permissions and ownership will still be enforced, however.

How about zeroing the file content [as is done here with the sources.list file](https://github.com/saltstack-formulas/apt-formula/blob/master/apt/repositories.sls#L13-L21) to address a similar issue (#19)

Something like

```yaml
etc/apt/preferences:
  file.managed:
    - mode: '0644'
    - user: root
    - group: root
  {% if remove_preferences %}
    - contents: ''
    - contents_newline: False
{% endif %}

Reply:
Current state is not changing the file in any case, so the `replace: False` seems ok. There currently is no source or contents that could possibly alter it...

I was just looking at the state itself...

Now, if we want to get coherent, yeah, having the `remove_preferences` and `remove_sources_list` act the same way, having the remove place it to empty file, and if not have the `replace: False` to avoid warning...

Reply:
> Now, if we want to get coherent, yeah, having the remove_preferences and remove_sources_list act the same way, having the remove place it to empty file, and if not have the replace: False to avoid warning...

@ze42 yes, the modification I suggested is to have a coherent behaviour in the formula. Notice that with the suggested changes (as is happening with the `/etc/apt/sources.list`), the `replace:` attribute is not needed, as the zeroing `contents:...` lines are added only if `remove_preferences: True`, Otherwise, the resource will just set permissions on the file (which will produce no warnings)

wdyt?